python/tank/platform/constants.py

```diff
@@ -51,8 +51,11 @@
 # app store approvals mode
 APP_STORE_QA_MODE_ENV_VAR = "TANK_QA_ENABLED"
 
-# use new experimental yaml parser
-PRESERVE_YAML_ENV_VAR = "TK_PRESERVE_YAML"
+# force use old, non-structure preseving parser
+USE_PYYAML_ENV_VAR = "TK_USE_PYYAML"
```

Comment:
So, in the same vein

TK_USE_LEGACY_YAML_PARSER feels like something that will be more obvious to a reader.